plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -183,10 +183,44 @@ public TokenBucket getUserRateLimiter(String modelId, String user) {
         return userRateLimiterMap.get(user);
     }
 
+    /**
+     * Set the ml interface for the model
+     *
+     * @param modelId model id
+     * @param modelInterface model interface
+     */
+    public synchronized void setModelInterface(String modelId, Map<String, String> modelInterface) {
+        log.debug("Setting ML Interface {} for Model {}", modelInterface, modelId);
```

Comment:
wondering if there is IT test for connector and check for the response to get connector, and see if the model interface is in the response body?

Reply:
Per Yaliang's request we are going to move the check to rest layer